### ISSUE TITLE
Rounding issues when creating a Duration from a double

### ISSUE THREAD

**Issue by Sibusten** [nodatime/nodatime#1695]
# Issue
Calling `Duration.FromSeconds(4.1)` creates a `Duration` with a time of `0:00:00:04.099999999` (`4099999999` nanoseconds).

https://dotnetfiddle.net/2tpYee

# Expected Behavior
A `Duration` with a time of `0:00:00:04.1` should be created.

# Information
Using NodaTime 3.1.0

**Comment by jskeet** [comment]
The literal 4.1 is *exactly* 4.0999999999999996447286321199499070644378662109375. So that many seconds is 4099999999.9999996447286321199499070644378662109375 nanoseconds, and Noda Time truncates the number of nanoseconds towards 0.

Basically I believe this is a matter of floating point representation rather than a fault in Noda Time - your code doesn't represent the value you think it does.

**Comment by Sibusten** [comment]
Yes, that's the underlying cause of the error. I have created a pull request with a proposed solution. Let me know what you think.
